src/core/net/nd6.hpp

```diff
@@ -525,6 +564,53 @@ class RaFlagsExtOption : public Option, private Clearable<RaFlagsExtOption>
 
 static_assert(sizeof(RaFlagsExtOption) == 8, "invalid RaFlagsExtOption structure");
 
+
+class ND6TxMessage {
```

Comment:
Regarding naming convention, we use `CamelCase` and treat abbreviations as a single word, so they also follow the `CamelCase` format and don't use capitals (examples `Icmp6Header`, `SrpClient`, `UdpSocket`, etc.)

So, I suggest `Nd6TxMessage`?

Or actually, since we're already under the `ot::Ip6::Nd` namespace, we could simply use `TxMessage`?